src/Field.php

```diff
@@ -59,13 +59,15 @@ class Field
      *
      * @return void
      */
-    public function __construct(Group $group, array $settings, Field $parent_field = null)
+    public function __construct(Group $group, array $settings, Field $parent_field = null )
     {
         $this->group = $group;
         $this->settings = $settings;
         $this->parent_field = $parent_field;
 
-        $this->setKey($settings['name']);
+        if ( 'tab' !== $settings['type'] ) {
```

Comment:
Interesting, why don't you want to set a key for tabs? Also, is there any other field we don't want to add a key to?

Reply:
My bad, of course tabs must have keys set too. But the name attribute should not be a requirement though. The same logic should be applied to the message type. Neither of these needs a static key, and therefor a name, because neither of them saves any data.

I've pushed another patch but only for the tab type, I think the logic should maybe be structured in another way to allow for more fields in the future not needing a static field key.

Reply:
Does this only apply to the tab and message types?

Reply:
Currently, yes. But there's nothing stopping Acf from adding more field types that doesn't save any data, and therefor doesn't require a static name/key.

Reply:
So, the tab and message fields only requires the type attribute?

Reply:
No, they also require the label attribute.

Reply:
Okay, great.